### DISCUSSION TITLE
In Blazor, how can I dynamically change an HTML tag?

### DISCUSSION THREAD

**Discussion by user72356** [dotnet/aspnetcore#26593] [Q&A] [answered]
Let's say I have the following markup in my Blazor component:

    <div @attributes=Attributes data-myattr="something">
        @ChildContent
    </div>

I'd like to provide a way for the parent component to determine which tag is going to be used in place of `<div>`. Something like:

    <@Tag @attributes=Attributes data-myattr="something">
        @ChildContent
    </@Tag>

With `@Tag` being a string Parameter. Of course that doesn't work. I'm aware of templates but it doesn't work for me because I want to be in control the structure of the tag, and add extra attributes to it. I just want to give the user a choice of which tag is going to be displayed.

**Comment by glararan** [comment, marked as answer]
You need to use **RenderTreeBuilder**. Blazor don't support at this moment what you asking for (I think so).
